### PR TITLE
Bluetooth: Update HCI driver interface

### PR DESCRIPTION
**Pull request by rugeGerritsen**
The new driver API was introduced in https://github.com/zephyrproject-rtos/zephyr/pull/72323.
    
The driver was updated similar to how to the zephyr link layer was updated.
Introducing another dts binding will be done seperately.

The intention is to allow running upstream tests with the SoftDevice Controller

### CODE REVIEW

**Review by bjda — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Comment by rugeGerritsen** [comment]
This PR triggered a HCI Reset Timeout in the Sidewalk CI after doing a bt_disable(). It looks like a bug in the host. Needs further investigation

File: samples/caf_sensor_manager/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 1]
```diff
@@ -36,6 +36,10 @@
 
 ipc0: &cpuapp_cpuppr_ipc {
```
Comment (hermabe) [inline samples/caf_sensor_manager/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:37]: This looks a bit odd, isnt this the ipc between APP and PPR? Same in other overlays for 54H.

Comment (rugeGerritsen) [reply samples/caf_sensor_manager/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:37]: @KAGA164 , can you provide some insight here?

**Review by anangl — APPROVED** [review]

**Review by hermabe — APPROVED** [review]

**Review by rick1082 — APPROVED** [review]

File: samples/bluetooth/nrf_dm/boards/nrf5340dk_nrf5340_cpuapp.overlay [hunk 2]
```diff
@@ -33,6 +33,11 @@
 			mbox-names = "tx", "rx";
 			role = "host";
 			status = "okay";
+
+			bt_hci_ipc0: bt_hci_ipc0 {
```
Comment (koffes) [inline samples/bluetooth/nrf_dm/boards/nrf5340dk_nrf5340_cpuapp.overlay:37]: Should this overlay also be added for the Audio DK?

Comment (rugeGerritsen) [reply samples/bluetooth/nrf_dm/boards/nrf5340dk_nrf5340_cpuapp.overlay:37]: That should be needed as the audio board reuses existing configurations. This sample deletes the IPC node which requires us to put it back.


The file `boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp.dts` includes `<nordic/nrf5340_cpuapp_qkaa.dtsi>` which includes `#include <nordic/nrf5340_cpuapp.dtsi>`  which again includes `dts/arm/nordic/nrf5340_cpuapp_ipc.dtsi` which defined the HCI IPC interface

**Comment by rugeGerritsen** [comment]
Rebased. Resolved a conflict in west.yml for the zephyr project

**Pull request merged**